ros_ws/src/autonomous/src/wall.cpp

```diff
@@ -10,19 +10,19 @@ Wall::Wall(float angle1, float angle2, float range1, float range2)
 
 float Wall::getAngle()
 {
-    return atan((m_range1 * cos(std::abs(m_angle1 - m_angle2)) - m_range2) / m_range1 *
-                sin(std::abs(m_angle1 - m_angle2)));
+    return atan((m_range1 * (float)cos(std::abs(m_angle1 - m_angle2)) - m_range2) / m_range1 *
+                (float)sin(std::abs(m_angle1 - m_angle2)));
 }
 
 float Wall::predictDistance(float distancce_to_current_position)
 {
     float angle = this->getAngle();
-    float currentWallDistance = m_range2 * cos(angle);
-    return currentWallDistance + distancce_to_current_position * sin(angle);
+    float currentWallDistance = m_range2 * (float)cos(angle);
+    return currentWallDistance + distancce_to_current_position * (float)sin(angle);
 }
 
 void Wall::draw(RvizGeometryPublisher& geometry, int id, std_msgs::ColorRGBA color)
 {
-    geometry.drawLine(id, createPoint(m_range1 * cos(m_angle1), m_range1 * sin(m_angle1), 0.0),
-                      createPoint(m_range2 * cos(m_angle2), m_range2 * sin(m_angle2), 0.0), color, 0.03);
+    geometry.drawLine(id, createPoint(m_range1 * (float)cos(m_angle1), m_range1 * (float)sin(m_angle1), 0.0),
```

Comment:
```suggestion
    geometry.drawLine(id, createPoint(m_range1 * std::cos(m_angle1), m_range1 * std::sin(m_angle1), 0.0f),
```
Because `cos(double)` from `math.h` is used, by replacing it with the C++ one from `cmath` (which is defined for `float`) no conversion from `double` to `float` should happen. Also added the `float` literal to the floating number.